### ISSUE TITLE
FormField-label used for aria-label

### ISSUE THREAD

**Issue by britt6612** [grommet/grommet#6483]
We need to look at defaulting `aria-label` as label if one of the following is not provided
 `aria-describedby`, `aria-description` or `aria-labeledBy`, `aria-label`.

**Comment by taysea** [comment]
In the cases where a visible text label is present (such as using `label` prop on FormField), I don't think we need `aria-label` correct? Is this only for cases where there is not visible labels?

**Comment by jcfilben** [comment]
Even if there is a visible label I think we still need an aria-label. (See #6466 for an example of this)

**Comment by halocline** [comment]
> Even if there is a visible label I think we still need an aria-label. (See #6466 for an example of this)

Is this true if there is an association between `for` and `id`? I noticed that the story above is not using `htmlFor` on the FormField and no `id` on the input.

**Comment by halocline** [comment]
> > Is this true if there is an association between for and id? I noticed that the story above is not using htmlFor on the FormField and no id on the input.
> 
> Good note, if we have `htmlFor` and `id` then we don't need an aria-label. I wonder which direction we should be encouraging, `aria-label` or `htmlFor`?
> 
> Maybe `htmlFor` should be used in most cases and `aria-label` should only be used if what the screenreader reads out should be different than the label text

Related https://github.com/grommet/grommet/pull/6540#discussion_r1049814124

**Comment by taysea** [comment]
> Maybe `htmlFor` should be used in most cases and `aria-label` should only be used if what the screenreader reads out should be different than the label text

Agree completely with this.

[first rule of ARIA](https://www.w3.org/TR/using-aria/#rule1) is: If you can use a native HTML element [[HTML51](https://www.w3.org/TR/using-aria/#bib-html51)] or attribute with the semantics and behavior you require already built in, instead of re-purposing an element and adding an ARIA role, state or property to make it accessible, then do so.

**Comment by britt6612** [comment]
closing since we all agree on the above comment.
